tasks/setup.yaml

```diff
@@ -0,0 +1,57 @@
+# Copyright 2024 Defense Unicorns
+# SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+
+tasks:
+  - name: restore-k3d-loadbalancer
+    Description: "Restores the k3d Load Balancer killed by the external-loadbalancer task"
+    actions:
+      - description: "Cleanup existing HAProxy containers"
+        cmd: |
+          docker kill haproxy || true
+          docker rm haproxy || true
+      - description: "Restore the k3d Load Balancer"
+        cmd: k3d cluster start uds
+
+  - name: external-loadbalancer
+    Description: "Setup HAProxy as an external load balancer for the test-infra namespace"
+    actions:
+      - description: "Cleanup existing HAProxy containers"
+        cmd: |
+          docker kill haproxy || true
+          docker rm haproxy || true
+      - description: "Prepare HAProxy certs"
+        cmd: |
+          yq -r '.tls.cert' uds-core/src/istio/values/config-tenant.yaml | base64 -d > tasks/haproxy.pem
+          yq -r '.tls.key' uds-core/src/istio/values/config-tenant.yaml | base64 -d > tasks/haproxy.pem.key
+          yq -r '.tls.cacert' src/test/cypress/certs/tls_cacert.yaml | base64 -d > tasks/haproxy-client-ca.pem
+      - description: "Sanity check HAProxy configuration"
+        cmd: |
+          TENANT_IP=$(uds zarf tools kubectl get svc tenant-ingressgateway -n istio-tenant-gateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}' || echo "unknown")
+          if [ "$TENANT_IP" != "172.18.0.201" ]; then
+            echo "\033[33m⚠️ Tenant Gateway external IP is not is not 172.18.0.201, got $TENANT_IP. The setup may not work correctly. \033[0m"
+          fi
```

Comment:
When I deploy `uds-k3d` I think my IP range changes from time to time (always in the `172.x` range). Could we just dynamically set/change this in the HAProxy config? Potentially just set it to an env in the config file, then pass that env into `docker run`?

Reply:
This was the initial approach I started with. It tuned out this stands in the way when debugging and restarting HAProxy frequently. In this scenario you often have the k3d Load Balancer turned down and there's no way to obtain the $TENANT_IP. This was the primary reason I decided to "fix" it in the configuration.

But I've made a compromise - I'm checking the IP and if it's indeed different - I'm calling sed and updating it.